processing/src/main/java/io/druid/query/QueryPlus.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query;
+
+import com.google.common.base.Preconditions;
+import io.druid.java.util.common.guava.Sequence;
+import io.druid.query.spec.QuerySegmentSpec;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+
+/**
+ * An immutable composite object of {@link Query} + extra stuff needed in {@link QueryRunner}s. This "extra stuff"
+ * is only {@link QueryMetrics} yet.
+ */
+public final class QueryPlus<T>
```

Comment:
Since `QueryRunner` is also parameterized only with `T`, the parameter of `QueryRunner.run()` could only be `QueryPlus<T, ? extends Query<T>>`, so it won't help much. Adding parameter to `QueryRunner` or `run()` method makes it too complex.

I think parametrization in this area pretty much doesn't work (but I don't think this is an issue), so I created `QueryPlus` to have the same parameter as `Query` and `QueryRunner`, to keep changes in this PR to minimum.